lib/rbtree/rb.c

```diff
@@ -39,14 +39,14 @@ static void set_child(struct rbnode *n, int side, void *val)
 		uintptr_t old = (uintptr_t) n->children[0];
 		uintptr_t new = (uintptr_t) val;
 
-		n->children[0] = (void *) (new | (old & 1ul));
+		n->children[0] = (void *) (new | (old & 1UL));
 	}
 }
 
 static enum rb_color get_color(struct rbnode *n)
 {
 	CHECK(n);
-	return ((uintptr_t)n->children[0]) & 1ul;
+	return ((uintptr_t)n->children[0]) & 1UL;
```

Comment:
MISRA cares about case of literal suffixes?  Seriously?

Reply:
yeah, the reason is that l can be mislead with 1